File: fastapi/entity/po/article.py
from sqlmodel import SQLModel, Field
from typing import Optional
from datetime import datetime

class Article(SQLModel, table=True):
    """文章实体类"""

    __tablename__ = "articles"
    id: Optional[int] = Field(default=None, primary_key=True, index=True)
    title: str = Field()
    content: str = Field()
    user_id: Optional[int] = Field(default=None)
    tags: Optional[str] = Field(default=None)
    status: Optional[str] = Field(default=None)
    create_at: Optional[datetime] = Field(default=None)
    update_at: Optional[datetime] = Field(default=None)
    views: Optional[int] = Field(default=0)
    sub_category_id: Optional[int] = Field(default=None)  # 子分类ID

File: gin/entity/po/article.go
package po

import "time"

type Article struct {
	ID            uint      `gorm:"column:id" json:"id"`
	Title         string    `gorm:"column:title" json:"title"`
	Content       string    `gorm:"column:content" json:"content"`
	UserID        uint      `gorm:"column:user_id" json:"userId"`
	SubCategoryID int       `gorm:"column:sub_category_id" json:"sub_category_id"`
	Tags          string    `gorm:"column:tags" json:"tags"`
	Status        int       `gorm:"column:status" json:"status"` // 1 表示已发布
	Views         int       `gorm:"column:views" json:"views"`   // 浏览量
	CreateAt      time.Time `gorm:"column:create_at" json:"createdAt"`
	UpdateAt      time.Time `gorm:"column:update_at" json:"updatedAt"`
}

func (Article) TableName() string {
	return "articles"
}
